interfaces_test.go

```diff
@@ -29,3 +38,57 @@ func TestNewRequest(t *testing.T) {
 		t.Errorf("Request mismatch (-want +got):\n%s", diff)
 	}
 }
+
+func TestMarshalStruct(t *testing.T) {
+	testSpan := &Span{
+		TraceID:      "d6c4f03650bd47699ec65c84352b6208",
+		SpanID:       "1cc4b26ab9094ef0",
+		ParentSpanID: "442bd97bbe564317",
+		Description:  `SELECT * FROM user WHERE "user"."id" = {id}`,
+		Op:           "db.sql",
+		Tags: map[string]string{
+			"function_name":  "get_users",
+			"status_message": "MYSQL OK",
+		},
+		StartTimestamp: time.Unix(0, 0).UTC(),
+		EndTimestamp:   time.Unix(5, 0).UTC(),
+		Status:         "ok",
+	}
+
+	testCases := []struct {
+		testName     string
+		sentryStruct interface{}
+	}{
+		{
+			testName:     "span",
+			sentryStruct: testSpan,
+		},
+	}
+
+	for _, test := range testCases {
+		test := test
+		t.Run(test.testName, func(t *testing.T) {
+			got, err := json.MarshalIndent(test.sentryStruct, "", "    ")
+			if err != nil {
+				t.Error(err)
+			}
+
+			golden := filepath.Join(".", "testdata", fmt.Sprintf("%s.golden", test.testName))
+			if *update {
+				err := ioutil.WriteFile(golden, got, 0600)
+				if err != nil {
+					t.Error(err)
+				}
+			}
+
+			want, err := ioutil.ReadFile(golden)
+			if err != nil {
+				t.Error(err)
```

Comment:
```suggestion
				t.Fatal(err)
```

If we cannot read the file, `want` has no bytes / nothing useful to compare with `got` later.